### PR TITLE
scraper: completely reworked asus crawler to use JSON endpoints, fixed ROG device handling

### PR DESCRIPTION
**Pull request by rhelmke**

### CODE REVIEW

**Review by jstucke — COMMENTED** [review]
Seems to work fine.

File: firmware/spiders/asus.py [hunk 1]
```diff
@@ -1,106 +1,95 @@
...
-        ro='Router (Gaming)'  # ROG Rapture
-    )
-    base_url = 'https://www.asus.com/de/Networking-IoT-Servers/{}/All-series/filter/'
-    start_urls = [
-        base_url.format('WiFi-Routers'),
-        base_url.format('Modem-Routers'),
-        base_url.format('WiFi-6')
-    ]
+
+    start_urls = ['https://odinapi.asus.com/recent-data/apiv2/SeriesFilterResult?SystemCode=asus&WebsiteCode=de&ProductLevel1Code=Networking-IoT-Servers&ProductLevel2Code=Wifi-Routers&PageSize=100&PageIndex=1&CategoryName=&SeriesName=ROG-Republic-of-Gamers,ASUS-Gaming-Routers,ASUS-WiFi-Routers&SubSeriesName=&Spec=&SubSpec=&Sort=Recommend&siteID=www&sitelang=']  # pylint: disable=line-too-long
```
[the remark this answers is not in the document]
Comment (rhelmke) [reply firmware/spiders/asus.py:19]: I thought about clickable links in text editors. But we can also split the lines, sure :-)

File: firmware/spiders/asus.py [hunk 2]
```diff
@@ -1,106 +1,95 @@
...
-            if url_redirect[-1] != '/':
-                continue
-            # selenium does not adhere to throttling settings, which is why we uniformly sleep to evade potential server-site throttling
-            sleep(uniform(0.5, 2.0))
-            yield Request(url=f'{url_redirect}HelpDesk_BIOS/', callback=self.parse_firmware, meta={'selenium': True})
+    def parse(self, response: Response, **kwargs) -> Generator[Request, None, None]:
+        try:
+            data = json.loads(response.body.decode())
+        except (UnicodeDecodeError, JSONDecodeError):
+            yield None
```
[the remark this answers is not in the document]
Comment (rhelmke) [reply firmware/spiders/asus.py]: Changed, as discussed.

**Review by jstucke — APPROVED** [review]

File: firmware/spiders/asus.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jstucke) [reply firmware/spiders/asus.py]: If it is useful to keep it in one line, that is also OK for me

**Pull request merged**